src/components/common/NetworkSelector/NetworkMultiSelector.tsx

```diff
@@ -0,0 +1,126 @@
+import useChains from '@/hooks/useChains'
+import { type ReactElement } from 'react'
+import { Checkbox, Autocomplete, TextField, Chip } from '@mui/material'
+import type { ChainInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import ChainIndicator from '../ChainIndicator'
+import css from './styles.module.css'
+import { Controller, useFormContext, useWatch } from 'react-hook-form'
+import { useRouter } from 'next/router'
+import { getNetworkLink } from '.'
+import useWallet from '@/hooks/wallets/useWallet'
+import { SetNameStepFields } from '@/components/new-safe/create/steps/SetNameStep'
+import { LATEST_SAFE_VERSION } from '@/config/constants'
+import { getSafeSingletonDeployment } from '@safe-global/safe-deployments'
+
+const NetworkMultiSelector = ({
+  name,
+  isAdvancedFlow = false,
+}: {
+  name: string
+  isAdvancedFlow?: boolean
+}): ReactElement => {
+  const { configs } = useChains()
+  const router = useRouter()
+  const isWalletConnected = !!useWallet()
+
+  const {
+    formState: { errors },
+    control,
+    getValues,
+    setValue,
+  } = useFormContext()
+
+  const selectedNetworks = useWatch({ control, name: SetNameStepFields.networks })
+
+  const handleDelete = (deletedChainId: string) => {
+    const currentValues: ChainInfo[] = getValues(name) || []
+    const updatedValues = currentValues.filter((chain) => chain.chainId !== deletedChainId)
+    updateSelectedNetwork(updatedValues)
+    setValue(name, updatedValues)
+  }
+
+  const updateSelectedNetwork = (chains: ChainInfo[]) => {
+    if (chains.length !== 1) return
+    const shortName = chains[0].shortName
+    const networkLink = getNetworkLink(router.pathname, router.query, shortName, isWalletConnected)
+    router.replace(networkLink)
+  }
+
+  const isOptionDisabled = (optionNetwork: ChainInfo) => {
```

Comment:
I am not sure if this function and the others could cause re-renders if they are not using `useCallback`